[18.0][MIG] account_statement_base: Migration to 18.0

xaviedoanhduy opened the pull request:
### depend on: 

- https://github.com/xaviedoanhduy/account-reconcile/pull/1

### change in v18

- `checked` was replace `to_checked` fields in commit: https://github.com/odoo/odoo/commit/242e1a04fc97131ccc28905de58011127f078e30

alexis-via committed:
[ADD] module account_statement_base

This module contains the views of account.bank.statement.line, which
have been removed from the account module in Odoo v16

OCA-git-bot committed:
[ADD] icon.png

etobella committed:
[IMP] account_reconcile_oca: Finish creation of module

* refactoring JS in order to add logic
* Improve views
* Minor fixes in Odoo code in order to fix all possible options

alexis-via committed:
account_statement_base: change licence to LGPL

The module account_statement_base will be used by the modules from
OCA/bank-statement-import which are LGPL (because they initially
came from Odoo official addons), so it cannot be AGPL.

OCA-git-bot committed:
account_statement_base 16.0.1.1.0

alexis-via committed:
account_statement_base: add development_status in manifest

OCA-git-bot committed:
account_statement_base 16.0.1.2.0

NL66278 committed:
[IMP] account_statement_base: add form to statement view_mode

OCA-git-bot committed:
account_statement_base 16.0.1.3.0

adrianojprado committed:
Added translation using Weblate (Portuguese (Brazil))

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (12 of 12 strings)

Translation: account-reconcile-16.0/account-reconcile-16.0-account_statement_base
Translate-URL: https://translation.odoo-community.org/projects/account-reconcile-16-0/account-reconcile-16-0-account_statement_base/pt_BR/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (12 of 12 strings)

Translation: account-reconcile-16.0/account-reconcile-16.0-account_statement_base
Translate-URL: https://translation.odoo-community.org/projects/account-reconcile-16-0/account-reconcile-16-0-account_statement_base/es/

remi-filament committed:
Added translation using Weblate (French)

remi-filament committed:
Translated using Weblate (French)

Currently translated at 100.0% (12 of 12 strings)

Translation: account-reconcile-16.0/account-reconcile-16.0-account_statement_base
Translate-URL: https://translation.odoo-community.org/projects/account-reconcile-16-0/account-reconcile-16-0-account_statement_base/fr/

rov-adhoc committed:
[IMP] account_statement_base: pre-commit auto fixes

rov-adhoc committed:
[MIG] account_statement_base: Migration to 17.0

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: account-reconcile-17.0/account-reconcile-17.0-account_statement_base
Translate-URL: https://translation.odoo-community.org/projects/account-reconcile-17-0/account-reconcile-17-0-account_statement_base/it/

lk-eska committed:
[IMP] account_statement_base: Set tree views

Make them listed and editable

sergiobstoj committed:
[IMP] account_statement_base: add smart button on view_bank_statement_form linking to journal items

[16.0][FIX] account_statement_base: Changes post OCA CR

sergio-teruel committed:
[IMP] account_statement_base: Add statement form view

xtanuiha committed:
Added translation using Weblate (Chinese (zh))

xtanuiha committed:
Added translation using Weblate (Chinese (Simplified) (zh_CN))

xtanuiha committed:
Translated using Weblate (Chinese (Simplified) (zh_CN))

Currently translated at 100.0% (12 of 12 strings)

Translation: account-reconcile-17.0/account-reconcile-17.0-account_statement_base
Translate-URL: https://translation.odoo-community.org/projects/account-reconcile-17-0/account-reconcile-17-0-account_statement_base/zh_CN/

JordiBForgeFlow committed:
[IMP] account_statement_base: add possibility to navigate from statement lines to the
associated journal entry through a button. Adds also the journal entry number
as an optional field

carlos-lopez-tecnativa committed:
[IMP] account_statement_base: Show action to create Statements

Before this commit, when a user clicked the "New Transaction" button, an error was raised. After this commit, statement lines are displayed to create new records.

etobella committed:
[IMP] account_statement_base: Add extra fields

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (15 of 15 strings)

Translation: account-reconcile-17.0/account-reconcile-17.0-account_statement_base
Translate-URL: https://translation.odoo-community.org/projects/account-reconcile-17-0/account-reconcile-17-0-account_statement_base/it/

JordiBForgeFlow committed:
[IMP] make the statements views editable=top by default
And add an icon to allow users to navigate to the statement lines.
Currently it's not possible to open the statement form view. When editing the
view an error occurs, related with the recomputation of running balances.

victoralmau committed:
[FIX] account_statement_base: Allow to create new manual transactions in tree view

Related to https://github.com/OCA/account-reconcile/pull/678

TT50970

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (17 of 17 strings)

Translation: account-reconcile-17.0/account-reconcile-17.0-account_statement_base
Translate-URL: https://translation.odoo-community.org/projects/account-reconcile-17-0/account-reconcile-17-0-account_statement_base/it/

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (19 of 19 strings)

Translation: account-reconcile-17.0/account-reconcile-17.0-account_statement_base
Translate-URL: https://translation.odoo-community.org/projects/account-reconcile-17-0/account-reconcile-17-0-account_statement_base/it/

clementthomas committed:
[FIX]account_statement_base: bug on new bank statement line fixed

victoralmau committed:
[FIX] account_statement_base: Add the possibility to create Bank Statements manually

Related to https://github.com/OCA/account-reconcile/pull/706

TT50906

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (20 of 20 strings)

Translation: account-reconcile-17.0/account-reconcile-17.0-account_statement_base
Translate-URL: https://translation.odoo-community.org/projects/account-reconcile-17-0/account-reconcile-17-0-account_statement_base/it/

victoralmau committed:
[FIX] account_statement_base: Allow remove lines from bank statment form view

Fixes https://github.com/OCA/account-reconcile/issues/697

TT50906

victoralmau committed:
[FIX] account_statement_base: Allow add lines from bank statment form view

TT50906

OCA-git-bot committed:
[BOT] post-merge updates

xaviedoanhduy committed:
[IMP] account_statement_base: pre-commit auto fixes

xaviedoanhduy committed:
[MIG] account_statement_base: Migration to 18.0

thienvh332 commented:
> change in v17
>> checked was replace to_checked fields in commit: https://github.com/odoo/odoo/commit/242e1a04fc97131ccc28905de58011127f078e30

It changed from version 18

xaviedoanhduy commented:
> > change in v17
> > > checked was replace to_checked fields in commit: [odoo/odoo@242e1a0](https://github.com/odoo/odoo/commit/242e1a04fc97131ccc28905de58011127f078e30)
> 
> It changed from version 18

ok thanks, wrong description

thienvh332 reviewed (approved):
LG overall,

Missing CREDITS.md

xaviedoanhduy committed:
[IMP] account_statement_base: no need to depend on account_reconcile_model_oca tests

victoralmau committed:
[FIX] account_statement_base: Allow statements to be created by clicking on the New button (only on cash ones).

If you access the list of statements from the Create button, you should be able to create new records.

TT51767

victoralmau committed:
[FIX] account_statement_base: Add default_statement_id to the context to create the statement lines linked to it

TT51767
